contracts/router/PowerTokenRouter.sol

```diff
@@ -199,6 +200,10 @@ contract PowerTokenRouter is UUPSUpgradeable, AccessControl {
             );
     }
 
+    function getImplementation() external view override returns (address) {
```

Comment:
Maybe better is use that method in abstract contract to not repeat this in every contract